JSTask4.js

```diff
@@ -0,0 +1,227 @@
+        location: "Kochi",
+        appraisal: 1,
+    },
+    {
+        first_name: "Dev",
+        last_name: "Raj",
+        age: 21,
+        position: "Architect",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 2,
+    },
+    {
+        first_name: "Thomas",
+        last_name: "Mathew",
+        age: 28,
+        position: "Architect",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 5,
+    },
+    {
+        first_name: "Deepa",
+        last_name: "Thomas",
+        age: 31,
+        position: "Senior Engineer",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 4,
+    },
+    {
+        first_name: "Antony",
+        last_name: "Thomas",
+        age: 31,
+        position: "Senior Engineer",
+        company: "ABC",
+        location: "Kochi",
+        appraisal: 4,
+    },
+];
+
+// Task 1
+// Create a new array named employees by merging ‘comp1’ and ‘comp2’ using a spread operator.
+
+let merge = [...comp1, ...comp2];
+console.log(merge);
+
+
+// Task 2
+/* Iter the whole employee list and log the employee's full name by creating a function.
+(Use spread operator, rest parameters, Object destructuring, array.join) */
+
+for(let i=0;i<merge.length;i++)
+{
+    for(employee in merge[i])
+    {
+        console.log(merge[i].first_name);
+        break;
+    }
+}
+function full_name(){
+    for(let i=0;i<merge.length;i++)
+    {
+        for(names in merge[i])
+        {
+            let merge1 = merge[i].first_name + " " + merge[i].last_name;
+            console.log(merge1);
+            break;
+        }
+    }
+}
+full_name();
+
+
+// Task 3
+// List all employees having the role 'Engineer' and age above '25'.
+
+let pos1 = merge.filter(pos => pos.position.indexOf('Engineer')!==-1);
+console.log(pos1);
+let pos2 = merge.filter(yr => yr.age > 25);
+console.log(pos2);
+
+let roleage = [];
+for(i in merge)
+{
+    if(merge[i].position == 'Engineer' && merge[i].age > 25)
+    {
+        roleage.push(merge[i].first_name);
+    }
+}
+console.log(roleage);
+
+
+// Task 4
+// Create a filter function to sort employees by position, by default the filter need to sort employee having the role 'Architect'.
+
+let merge2 = merge.filter(({position}) => position === "Architect");
+console.log(merge2);
+
+
+// Task 5
+// Create a function for checking whether all the employees got appraisals once.
+
+function gotappraisal(merge)
+{
+    for(i in merge)
+    {
+        if(merge[i].appraisal == 0)
+        {
+            return console.log(merge[i].first_name, merge[i].last_name);  // has not got any appraisal at all
+        }
+    }
+}
+gotappraisal(merge);
+
+
+// Task 6
+// Create a function for checking whether any of the employees got an appraisal.
+
+function gotappraisal2(merge)
+{
+    let merge3 = merge.filter(({appraisal})=>appraisal > 0);
+    if(merge3.length > 0)
+    {
+        return console.log(merge3.length);
+    }
+}
+gotappraisal2(merge);
+
+
```

Comment:
The result includes the number of employees with appraisals. The expected output should be true or false.